src/Component/Filter/ComparisonFilter/ComparisonFilter.tsx

```diff
@@ -217,27 +210,17 @@ class ComparisonFilterUi extends React.Component<ComparisonFilterProps, Comparis
    * Stores the appropriate filter value as member.
    */
   onValueChange = (newValue: string | number | boolean) => {
-
-    this.setState({value: newValue});
-
-    // (re)create the ComparisonFilter object if all info are collected
-    if (this.state.operator && this.state.attribute && newValue) {
-      const compFilter: ComparisonFilter = [
-        this.state.operator, this.state.attribute, newValue
-      ];
-      this.setState({filter: compFilter});
-
-      this.props.onFilterChange(compFilter);
-    }
+    let filter: GsComparisonFilter = _cloneDeep(this.state.filter);
+    filter[2] = newValue;
```

Comment:
don't we need to clone?

Reply:
The `cloneDeep` is just need when manipulating the "immutable" state.